R/utils.R

```diff
@@ -299,7 +299,34 @@ create_app_id <- function(data, modules) {
   data <- if (inherits(data, "teal_data")) {
     as.list(data@env)
   } else if (inherits(data, "teal_data_module")) {
-    body(data$server)
+    deparse1(body(data$server))
   }
+  modules <- lapply(flatten_modules(modules), function(x) lapply(x, defunction))
```

Comment:
Given that modules always has `teal_modules` class due to assertion above, this is safe to do

```suggestion
  modules <- lapply(
    unlist(lapply(modules$children, flatten_modules), recursive = FALSE),
    defunction
  )
```

Or

```suggestion
  modules <- lapply(modules, defunction)
```